### PR TITLE
Frag - Implement Arma v2.18 changes from #10157

### PR DESCRIPTION
**Pull request by lambdatiger**
This PR requires #10157 to be merged.
**When merged this pull request will:**
- ~~Use `_instigator` projectile "HitPart" event handler argument instead of retrieving `shotParents`.~~
- Use new syntax of `nearEntities` to avoid needing to sort/filter the fragment target arrays.

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by PabstMirror** [comment]
Readable diff (I think) - https://github.com/lambdatiger/ace_frag-overhaul/compare/lambdatiger:ace_frag-overhaul:Event-hanlder-replacement...eventHandler-replace-2.18

**Comment by johnb432** [comment]
@lambdatiger Can you handle the merge conflicts?

**Comment by lambdatiger** [comment]
There's one more 2.18 change that came from #10247 that isn't present in this PR yet. I'll test it again later this week before I commit it.

**Comment by johnb432** [comment]
> There's one more 2.18 change that came from #10247 that isn't present in this PR yet. I'll test it again later this week before I commit it.

There's no hurry, given that 2.18 won't release until sometime in october (and I doubt we will release a day 1 patch).

**Comment by johnb432** [comment]
> There's no hurry, given that 2.18 won't release until sometime in october (and I doubt we will release a day 1 patch).

That aged like milk...

**Comment by lambdatiger** [comment]
I'll run through it tomorrow and let you know if it's good to go.

**Comment by lambdatiger** [comment]
Alright, new feature isn't easily added. Testing seems to work like it should, if someone else wants to try feel free. I'll address any other changes in #9728.

File: addons/frag/functions/fnc_frago.sqf [hunk 1]
```diff
@@ -39,105 +39,94 @@ _shellType call FUNC(getFragInfo) params ["_fragRange", "_fragVelocity", "_fragT
...
+TRACE_1("",_randomCount);
+private _sectorSize = 360 / (_randomCount max 1);
+
+for "_i" from 1 to _randomCount do {
+    // Distribute evenly
+    private _sectorOffset = 360 * (_i - 1) / (_randomCount max 1);
+    private _randomDir = random (_sectorSize);
+    private _vectorDir = [cos (_sectorOffset + _randomDir), sin (_sectorOffset + _randomDir), sin (30 - (random 45))];
 
         private _fragObjSpeed = _fragVelocity * (1 - random 0.5);
```
Comment (johnb432) [inline addons/frag/functions/fnc_frago.sqf:131]: Correct indentation.

File: addons/frag/functions/fnc_frago.sqf [hunk 2]
```diff
@@ -146,13 +135,13 @@ if (_targets isNotEqualTo []) then {
...
+    #ifdef DEBUG_MODE_DRAW
+    [_fragObj, "blue", true] call FUNC(dev_trackObj);
+    #endif
+    INC(_fragCount);
 };
 
 TRACE_1("total created",_fragCount);
 
 END_COUNTER(frago);
+_fragCount
```
Comment (johnb432) [inline addons/frag/functions/fnc_frago.sqf:147]:
```suggestion
_fragCount // return
```

File: addons/frag/functions/fnc_frago.sqf [hunk 3]
```diff
@@ -39,105 +39,94 @@ _shellType call FUNC(getFragInfo) params ["_fragRange", "_fragVelocity", "_fragT
...
-// Post 2.18 change - uncomment line 43, modify lines 45, and remove lines 44, 51-57, 64-66
-// private _targets = [ASLToAGL _fragPosAGL, _fragRange, _fragRange, 0, false, _fragRange] nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], false, true, true];
-private _objects = _fragPosAGL nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], _fragRange];
-if (_objects isEqualTo []) exitWith {
+
+// compile possible targets including units, vehicles, and crews
+private _targets = [_fragPosAGL, _fragRange, _fragRange, 0, false, _fragRange] nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], false, true, true];
+if (_targets isEqualTo []) exitWith {
     TRACE_2("No nearby targets",_fragPosAGL,_fragRange);
     0
```
Comment (johnb432) [inline addons/frag/functions/fnc_frago.sqf:47]:
```suggestion
    0 // return
```

File: addons/frag/functions/fnc_frago.sqf [hunk 4]
```diff
@@ -39,105 +39,94 @@ _shellType call FUNC(getFragInfo) params ["_fragRange", "_fragVelocity", "_fragT
 
 private _fragPosAGL = ASLToAGL _fragPosASL;
 TRACE_5("fragValues",_fragPosASL,_fragPosAGL,_fragRange,_fragVelocity,_metalMassModifier);
-// Post 2.18 change - uncomment line 43, modify lines 45, and remove lines 44, 51-57, 64-66
-// private _targets = [ASLToAGL _fragPosAGL, _fragRange, _fragRange, 0, false, _fragRange] nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], false, true, true];
-private _objects = _fragPosAGL nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], _fragRange];
-if (_objects isEqualTo []) exitWith {
+
+// compile possible targets including units, vehicles, and crews
```
Comment (johnb432) [inline addons/frag/functions/fnc_frago.sqf:43]:
```suggestion
// Compile possible targets including units, vehicles, and crews
```

**Review by johnb432 — CHANGES_REQUESTED** [review]
Other than that, LGTM

File: addons/frag/functions/fnc_frago.sqf (hunk 1 above)
Comment (lambdatiger) [reply addons/frag/functions/fnc_frago.sqf]: Done in https://github.com/acemod/ACE3/pull/10313/commits/da81d29693911979e7a23cf3e7256204a09120ed

**Review by johnb432 — APPROVED** [review]

**Pull request merged**